plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java

```diff
@@ -126,6 +127,7 @@
     private static final Pattern BUCKET_FROM_FILENAME_PATTERN = Pattern.compile("(0[0-9]+)_.*");
 
     private final Set<HiveFileWriterFactory> fileWriterFactories;
+    private final TrinoFileSystemFactory fileSystemFactory;
```

Comment:
Would be good to remove field `private final HdfsEnvironment hdfsEnvironment;` from this class.
1. I think `TrinoFileSystem` requires `exits` method (can be added)

There is also code in constructor:
```
        // make sure the FileSystem is created with the correct Configuration object
        try {
            hdfsEnvironment.getFileSystem(session.getIdentity(), writePath, conf);
        }
```
I think it's a sanity check.

Reply:
> I think TrinoFileSystem requires exits method (can be added) 

I think it can be done like this:
```
fileSystemFactory.create(session).newInputFile("path").exists()
```

Reply:
Right!.

Let's do it then, then we can remove `HdfsEnvironment hdfsEnvironment` field